### PR TITLE
Firebase integration + Analytics

### PR DESCRIPTION
**Pull request by wangannie**
[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

# What's new in this PR

[//]: # "Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
* Integrated Firebase setup for Firebase Analytics
* Added a few basic analytics events tracking just for testing: 
  * When resources external links are opened
  * Logs user IDs
  * Navigation screen changes

### Online sources
[//]: # "Optional - copy links to any tutorial or documentation that was useful to you when working on this PR"
Set up navigation screen tracking here:  https://reactnavigation.org/docs/screen-tracking/
More about firebase analytics: https://docs.expo.io/versions/latest/sdk/firebase-analytics/

### Related PRs

[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

## How to review

[//]: # "The order in which to review files and what to expect when testing locally"
* See if events log in the Firebase Console

## Next steps

[//]: # "What's NOT in this PR, doesn't work yet, and/or still needs to be done"
* Add more comprehensive analytics events
* Integrate Firebase phone auth
## Tests Performed, Edge Cases

[//]: # "Hopefully we will add a testing suite/CI soon, but until then note down the steps you took to test locally"

### Screenshots
image

CC: @anniero98 

[//]: # "This tags in both Annies as a default. Feel free to change, or add on anyone who you should be in on the conversation."

### CODE REVIEW

File: screens/auth/LogInScreen.js [hunk 1]
```diff
@@ -44,6 +45,7 @@ export default class LogInScreen extends React.Component {
   // From SignUpScreen. Sign in function. It sets the user token in local storage
   // to be the user ID and then navigates to homescreen.
   _asyncLogIn = async customerId => {
+    await Analytics.setUserId(customerId);
```
Comment (FranCogMents) [inline screens/auth/LogInScreen.js:48]: suggestion is to move this right above where Sentry registers a use on successful login  on line 114
```
Sentry.configureScope(scope => {
```

File: screens/auth/SignUpScreen.js [hunk 2]
```diff
@@ -88,6 +89,7 @@ export default class SignUpScreen extends React.Component {
   // Sign up function. It sets the user token in local storage
   // to be the fname + lname and then navigates to homescreen.
   _asyncSignUp = async customerId => {
+    await Analytics.setUserId(customerId);
```
Comment (FranCogMents) [inline screens/auth/SignUpScreen.js:92]: same suggestion from login, to move setting the customerId above where Sentry registers the customer on a successful signup above line 140
```
   // if signup works, register the user
   await Analytics.setUserId(customer.id);
      Sentry.configureScope(scope => {
```
